src/client.coffee

```diff
@@ -15,289 +15,252 @@ parser = new xml2js.Parser
 
 ###
   3Scale client API
-  Parameter:
-    provider_key {String} Required
-    default_host {String} Optional
+  Parameter: 
+    provider_key {String} Required if you don't use service_token in the method. 
+    Options: 
+      host {String} Optional
+      port {number} Optional
+  
   Example:
     Client = require('3scale').Client
-    client = new Client(provider_key, [default_host], [default_port])
+    
+    client = new Client(provider_key, options)    //If you use your provider_key
+    or 
+    client = new Client(options)                  //If you use a service_token in the method
 ###
 
 
 module.exports = class Client
   DEFAULT_HEADERS: { "X-3scale-User-Agent": "plugin-node-v#{VERSION}" }
 
-  constructor: (provider_key, host, port) ->
-    chosenHost = host || "su1.3scale.net";
-    chosenPort = port || 443;
-    
-    unless provider_key?
-      throw new Error("missing provider_key")
-   
-    @provider_key = provider_key
-    @host = chosenHost
-    @port = chosenPort
+
+  constructor: (provider_key, options) ->
+    if typeof provider_key is 'object' and options is undefined
+      opts = provider_key
+    else
+      @provider_key = provider_key
```

Comment:
Your deprecation warning should be here not below

`console?.warn?('provider_key is deprecated in favor of service_token with service_id. The release of this client will default to use service_token')`